src/app/public/modules/ag-grid/ag-grid.service.ts

```diff
@@ -47,6 +47,31 @@ function autocompleteFormatter(params: ValueFormatterParams): string | undefined
   return params.value && params.value.name;
 }
 
+function dateComparator(date1: any, date2: any): number {
+  let date1value = date1;
+  let date2value = date2;
+
+  if (typeof date1value === 'string') {
+    date1value = new Date(date1);
+  }
+
+  if (typeof date2value === 'string') {
+    date2value = new Date(date2);
+  }
+
+  if (date1value && date2value && date1value > date2value) {
```

Comment:
Since the first three `if`s test the same condition, this could be simplified by pulling that condition out and having a nested `if`. You can also get rid of the `else`s since the code in `if`s return values.

```
if (date1value && date2value) {
  if (date1value > date2value) {
    return 1;
  }

  if (date1value < date2value) {
    return -1;
  }

  return 0;
}

return date1value ? 1 : -1;
```